accepted/single-file/design.md

```diff
@@ -66,225 +32,254 @@ Publishing to a single file can be triggered by adding the following property to
-
-  * Publish directory contains: `HelloWorld.exe` `HelloWorld.pdb`
-
-  * `HelloWorld.dll`, `HelloWorld.deps.json`, and `HelloWorld.runtimeconfig.json` are embedded within `HelloWorld.exe`.
-
-* Run: `HelloWorld.exe`
-
-  * The app runs completely from the single-file, without the need for intermediate extraction to file.
-
-#### Self-Contained HelloWorld
-
-- Normal publish: `dotnet publish -r win10-x64`
-
-  * Publish directory contains 221 files including the host, the app, configuration files, the PDB-file and the runtime.
-
-- Single-file publish: `dotnet publish -r win10-x64 /p:PublishSingleFile=true`
-
-  - Publish directory contains: `HelloWorld.exe` `HelloWorld.pdb`
-  - The remaining 219 files are embedded within the host `HelloWorld.exe`.
-
-- Run: `HelloWorld.exe`
+To summarize, here's the overall experience for creating a HelloWorld single-file app:
 
-  * The bundled app and configuration files are processed directly from the bundle.
-  * Remaining 216 files will be extracted to disk at startup. 
-  * If reuse of extracted files is enabled, subsequent runs of the app may skip the extraction step.
+* Create a new HelloWorld app: `HelloWorld$ dotnet new console`
 
+* Framework Dependent Publish
 
-Most applications are expected to work without any changes. However, apps with a strong expectation about absolute location of dependent files may need to be made aware of bundling and extraction aspects of single-file publishing. No difference is expected with respect to debugging and analysis of apps.
+  * Normal publish: `dotnet publish` 
+    * Published files: `HelloWorld.exe`, `HelloWorld.dll`, `HelloWorld.deps.json`, `HelloWorld.runtimeconfig.json`, `HelloWorld.pdb`
 
-## Related Work
+  * Single-file publish: `dotnet publish -r win10-x64 --self-contained=false /p:PublishSingleFile=true`
+    * Published files: `HelloWorld.exe`, `HelloWorld.pdb`
 
-* [Mono/MkBundle](https://github.com/mono/mono/blob/master/mcs/tools/mkbundle) 
-* [Fody.Costura](https://github.com/Fody/Costura)
-* [Warp](https://github.com/dgiagio/warp)
-* [dotnet-warp](https://github.com/Hubert-Rybak/dotnet-warp)
-* [BoxedApp](https://docs.boxedapp.com/index.html)
-* [LibZ](https://github.com/MiloszKrajewski/LibZ)
+* Self-Contained Publish
 
+  * Normal publish: `dotnet publish -r win10-x64`
+      * Published files: `HelloWorld.exe`, `HelloWorld.pdb`, and 224 more files 
+  * Single-file publish Linux: `dotnet publish -r linux-x64 /p:PublishSingleFile=true`
+      * Published files: `HelloWorld.exe`, `HelloWorld.pdb`
+  * Single-file publish Windows: `dotnet publish -r win10-x64 /p:PublishSingleFile=true`
+      * Published files: `HelloWorld.exe`, `HelloWorld.pdb`, `coreclr.dll`, `clrjit.dll`, `clrcompression.dll`,  `mscordaccore.dll`
+  * Single-file publish Windows with Extraction: `dotnet publish -r win10-x64 /p:PublishSingleFile=true /p:IncludeContentInSingleFile=true /p:ExtractContentsFromSingleFile=true`
```

Comment:
Are we going to also keep the publish with Extractions for Unix?

Reply:
If we are dropping it, we should explain why.

Reply:
The option to `ExtractContentsFromSingleFile` is available for Unix too, mainly for user's own native binaries that may be included with the app. I did not include it here, because it doesn't apply to HelloWorld app.